board/sunnypilot/safety_mads.h

```diff
@@ -111,28 +111,32 @@ static void m_update_control_state(void) {
   // Primary control blockers - these prevent any further control processing
   if (m_mads_state.acc_main.transition == MADS_EDGE_FALLING) {
     mads_exit_controls(MADS_DISENGAGE_REASON_ACC_MAIN_OFF);
-    process_controls = false; // No matter what, no further control processing on this cycle
+    allowed = false;  // No matter what, no further control processing on this cycle
   }
 
   // Secondary control conditions - only checked if primary conditions don't block further control processing
-  if (process_controls && m_mads_state.disengage_lateral_on_brake) {
+  if (allowed && m_mads_state.disengage_lateral_on_brake) {
     // Brake rising edge immediately blocks controls
+    // Brake release might request controls if brake was the ONLY reason for disengagement
     if (m_mads_state.braking.transition == MADS_EDGE_RISING) {
       mads_exit_controls(MADS_DISENGAGE_REASON_BRAKE);
-      process_controls = false;
-    }
-    // Brake release might request controls if brake was the ONLY reason for disengagement
-    else if ((m_mads_state.braking.transition == MADS_EDGE_FALLING) && (m_mads_state.current_disengage.active_reason == MADS_DISENGAGE_REASON_BRAKE) && (m_mads_state.current_disengage.pending_reasons == MADS_DISENGAGE_REASON_BRAKE)) {
+      allowed = false;
+    } else if ((m_mads_state.braking.transition == MADS_EDGE_FALLING) &&
+               (m_mads_state.current_disengage.active_reason == MADS_DISENGAGE_REASON_BRAKE) &&
+               (m_mads_state.current_disengage.pending_reasons == MADS_DISENGAGE_REASON_BRAKE)) {
       m_mads_state.controls_requested_lat = true;
     } else if (m_mads_state.braking.current) {
-      process_controls = false;
+      allowed = false;
     } else {
-      // Do nothing
     }
   }
 
+  return allowed;
+}
+
+static void m_publish_state(bool allowed) {
```

Comment:
If you weren't allowed, why would you call this method?  and if you are allowed, do you really need to pass it as argument?

Reply:
And following that line of why do you need an argument on your method if it makes no sense, what about getting the allowed from `m_update_control_state()` directly inside of this method?